front/src/modules/trainschedule/components/ImportTimetableItem/ImportTimetableItemConfig.tsx

```diff
@@ -269,18 +291,189 @@ const ImportTimetableItemConfig = ({
 
       const trainSchedule: ImportedTrainSchedule = {
         trainNumber,
-        rollingStock: rollingStockViriato, // RollingStocks in viriato files rarely have the correct format
+        rollingStock: rollingStockXml, // RollingStocks in xml files rarely have the correct format
         departureTime: `${startDate} ${firstDepartureTimeformatted}`,
         arrivalTime: `${startDate} ${lastDepartureTimeformatted}`,
         departure: '', // Default for testing
         steps: adaptedSteps,
       };
-
+      trainSchedulesByTrainPartId[trainPartId] = trainSchedule;
       trainSchedules.push(trainSchedule);
     });
+
+    trainGroups.forEach((trainGroup) => {
+      const pacedTrainId = trainGroup.getAttribute('id')!;
+
+      const trainRefs = Array.from(trainGroup.getElementsByTagName('trainRef'));
+      pacedTrains[pacedTrainId] = trainRefs
+        .map((trainRef) => {
+          const trainId = trainRef.getAttribute('ref');
+          const trainElement = xmlDoc.querySelector(`train[id="${trainId}"]`);
+
+          const trainPartRef = trainElement?.querySelector('trainPartRef')?.getAttribute('ref');
+
+          return trainPartRef ? trainSchedulesByTrainPartId[trainPartRef] : undefined;
+        })
+        .filter((schedule): schedule is ImportedTrainSchedule => schedule !== undefined);
+    });
+
+    const getRelativeStepsTimeAndNames = (trainSchedule: ImportedTrainSchedule): string => {
+      const departureTime = new Date(trainSchedule.departureTime);
+
+      const stepsWithRelativeTimesAndNames = trainSchedule.steps.map((step) => {
+        const arrival = new Date(step.arrivalTime);
+        const departure = new Date(step.departureTime);
+
+        return {
+          name: step.name,
```

Comment:
Using `name` here as a unique identifier sounds fragile: it's a human-readable string. Could we use `uic` and `trigram` instead?